client/scripts/views/components/search_bar.ts

```diff
@@ -225,295 +269,352 @@ const getBalancedContentListing = (unfilteredResults: any[], types: SearchType[]
+          : !results || results?.length === 0
+            ? app.search.getByQuery(searchQuery)?.loaded
+              ? m(ListItem, [m(emptySearchPreview, { searchTerm })])
+              : m(ListItem, { label: m(Spinner, { active: true }) })
+            : vnode.state.isTyping
+            ? m(ListItem, { label: m(Spinner, { active: true }) })
+            : results
+      ])
 
     const chainOrCommIcon = app.activeId()
       ? app.activeChainId()
         ? m(ChainIcon, {
-          size: 18,
-          chain: app.chain.meta.chain,
-        })
+            size: 18,
+            chain: app.chain.meta.chain,
+          })
         : m(CommunityIcon, {
-          size: 18,
-          community: app.community.meta,
-        })
+            size: 18,
+            community: app.community.meta,
+          })
       : null;
     const cancelInputIcon = vnode.state.searchTerm
       ? m(Icon, {
-        name: Icons.X,
-        onclick: () => {
-          const input = $('.SearchBar').find('input[name=search');
-          input.val('');
-          vnode.state.searchTerm = '';
-        } })
+          name: Icons.X,
+          onclick: () => {
+            const input = $('.SearchBar').find('input[name=search');
+            input.val('');
+            vnode.state.searchTerm = '';
+          },
+        })
       : null;
-    return m(ControlGroup, {
-      class: 'SearchBar'
-    }, [
-      m(Input, {
-        name: 'search',
-        placeholder: 'Type to search...',
-        autofocus: false, // !isMobile,
-        fluid: true,
-        tabIndex: -10,
-        contentLeft: m(SearchIcon, {
-          isMobile,
-          intent: IconIntent.Primary,
-        }),
-        contentRight: cancelInputIcon || chainOrCommIcon,
-        defaultValue: m.route.param('q') || vnode.state.searchTerm,
-        value: vnode.state.searchTerm,
-        autocomplete: 'off',
-        oncreate: (e) => {
-          initializeSearch();
-        },
-        onclick: async (e) => {
-          vnode.state.focused = true;
-        },
-        oninput: (e) => {
-          e.stopPropagation();
-          vnode.state.isTyping = true;
-          vnode.state.searchTerm = e.target.value?.toLowerCase();
-          if (vnode.state.hideResults) {
+
+    const searchIcon = vnode.state.searchTerm
+      ? m(Icon, {
+          name: Icons.CORNER_DOWN_LEFT,
+          onclick: () => {
+            executeSearch(vnode.state.searchQuery)
+          },
+        })
+      : null;
+
+    return m(
+      ControlGroup,
+      {
+        class: 'SearchBar',
+      },
+      [
+        m(Input, {
+          name: 'search',
+          placeholder: 'Type to search...',
+          autofocus: false, // !isMobile,
+          fluid: true,
+          tabIndex: -10,
+          contentLeft: m(SearchIcon, {
+            isMobile,
+            intent: IconIntent.Primary,
+          }),
+          contentRight: vnode.state.searchTerm ? m(ControlGroup, {}, [searchIcon, cancelInputIcon]) : chainOrCommIcon,
```

Comment:
Might be nice to switch the X and <-- (cancel/search) icons so that the X is on the right side.

Reply:
Done!